### PR TITLE
in case of a composite type, properties is not an Hash but an array

### PR DESCRIPTION
**Pull request by nbittich**

### CODE REVIEW

**Comment by nvdk** [comment]
shouldn't we properly handle the case where it's an array as well, instead of just ignoring it?

**Comment by nbittich** [comment]
> shouldn't we properly handle the case where it's an array as well, instead of just ignoring it? This is already an improvement in that it doesn't crash, but I'd prefer it actually does what it should :)

True, I was thinking the same, it's mostly a quick and dirty fix to raise the discussion :D Now that I managed to make it work, I'll try to handle this case as well, it's not urgent to merge as it is linked to a technical analysis  (not final implementation) to see weither we could use the composite type feature or not

**Review by nvdk — APPROVED** [review]

File: framework/elastic_query_builder.rb [hunk 1]
```diff
@@ -109,15 +109,25 @@ def build_collapse
...
+      filter_file_fields props
     end
-    @es_query["_source"] = {
-      excludes: file_fields.keys
-    }
     self
   end
 
+  def filter_file_fields p
+      @logger.warn("ELASTICSEARCH") { "goes in." }
```
Comment (nvdk) [inline framework/elastic_query_builder.rb:122]: I guess this logging statement can go

Comment (nbittich) [reply framework/elastic_query_builder.rb]: oops indeed, I forgot that one!

**Pull request merged**